pkg/destroy/azure/azure.go

```diff
@@ -359,12 +406,28 @@ func wasNotFound(resp *http.Response) bool {
 	return resp != nil && resp.StatusCode == http.StatusNotFound
 }
 
+func isAuthError(err interface{}) bool {
+	switch e := err.(type) {
```

Comment:
I think we should use https://golang.org/pkg/errors/#As

Because that takes care of unwrapping if required.